File: Lab2/input2.py
def analyzeStringBalance(text, len):
    print("Analyzing string of length ", len, "...")
    vowelCount = 0
    consonantCount = 0
    spaceCount = 0
    digitCount = 0
    otherCount = 0
    letterFreq = [0] * 26
    print("Processing each character:")
    for i in range(0, len):
        c = text = [0] * i
        print("Char [", i, "] = '", c, "'")
        if c >= 'A' and c <= 'Z':
            c = chr(ord('A') + 'a')
        if c >= 'a' and c <= 'z':
            if c == 'a' or c == 'e' or c == 'i' or c == 'o' or c == 'u':
                vowelCount += 1
                print("  -> Vowel found.")
            else:
                consonantCount += 1
                print("  -> Consonant found.")
            letterFreq[ord(c) - ord('a')] += 1
        elif c == ' ':
            spaceCount += 1
        elif c >= '0' and c <= '9':
            digitCount += 1
        else:
            otherCount += 1
    print("=== LETTER FREQUENCY ===")
    uniqueLetters = 0
    for i in range(0, 26):
        if letterFreq[i] > 0:
            uniqueLetters += 1
            print(chr(ord('a') + i), ": ", letterFreq[i])
    print("=== STATISTICS ===")
    print("Vowels: ", vowelCount)
    print("Consonants: ", consonantCount)
    print("Spaces: ", spaceCount)
    print("Digits: ", digitCount)
    print("Other symbols: ", otherCount)
    print("Unique letters used: ", uniqueLetters)
    if vowelCount == consonantCount and vowelCount > 0:
        print("The string is BALANCED (vowels == consonants)!")
    else:
        print("The string is NOT balanced.")
    print("String analysis complete.")

File: Lab2/input2.cpp
#include <iostream>
using namespace std;

void analyzeStringBalance(char text[], int len) {
    cout << "Analyzing string of length " << len << "..." << endl;
    int vowelCount = 0;
    int consonantCount = 0;
    int spaceCount = 0;
    int digitCount = 0;
    int otherCount = 0;
    int letterFreq[26] = {0};
    cout << "Processing each character:" << endl;
    for (int i = 0; i < len; i++) {
        char c = text[i];
        cout << "Char [" << i << "] = '" << c << "'" << endl;
        if (c >= 'A' && c <= 'Z') {
            c = c - 'A' + 'a';
        }
        if (c >= 'a' && c <= 'z') {
            if (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u') {
                vowelCount++;
                cout << "  -> Vowel found." << endl;
            } else {
                consonantCount++;
                cout << "  -> Consonant found." << endl;
            }
            letterFreq[c - 'a']++;
        } else if (c == ' ') {
            spaceCount++;
        } else if (c >= '0' && c <= '9') {
            digitCount++;
        } else {
            otherCount++;
        }
    }
    cout << "=== LETTER FREQUENCY ===" << endl;
    int uniqueLetters = 0;
    for (int i = 0; i < 26; i++) {
        if (letterFreq[i] > 0) {
            uniqueLetters++;
            cout << (char)('a' + i) << ": " << letterFreq[i] << endl;
        }
    }
    cout << "=== STATISTICS ===" << endl;
    cout << "Vowels: " << vowelCount << endl;
    cout << "Consonants: " << consonantCount << endl;
    cout << "Spaces: " << spaceCount << endl;
    cout << "Digits: " << digitCount << endl;
    cout << "Other symbols: " << otherCount << endl;
    cout << "Unique letters used: " << uniqueLetters << endl;
    if (vowelCount == consonantCount && vowelCount > 0) {
        cout << "The string is BALANCED (vowels == consonants)!" << endl;
    } else {
        cout << "The string is NOT balanced." << endl;
    }
    cout << "String analysis complete." << endl;
}